src/main/java/org/openjdk/shenandoah/ShenandoahVisualizer.java

```diff
@@ -560,6 +604,12 @@ public synchronized void run() {
                     lastSnapshots.removeFirst();
                 }
                 frame.repaint();
+                if (popups != null) {
```

Comment:
This code to render popups is duplicated in quite a few places. Could we have a `repaintPopups` method somewhere that does this?